packages/core-modules/contracts/submodules/election/ElectionTally.sol

```diff
@@ -0,0 +1,109 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ElectionBase.sol";
+
+contract ElectionTally is ElectionBase {
+    using SetUtil for SetUtil.AddressSet;
+
+    function _evaluateNextBallotBatch(uint numBallots) internal {
+        if (numBallots == 0) {
+            numBallots = _electionStore().settings.defaultBallotEvaluationBatchSize;
+        }
+
+        ElectionData storage election = _getCurrentElection();
+        uint totalBallots = election.ballotIds.length;
+
+        uint firstBallotIndex = election.numEvaluatedBallots;
+
+        uint lastBallotIndex = firstBallotIndex + numBallots;
+        if (lastBallotIndex >= totalBallots) {
+            lastBallotIndex = totalBallots - 1;
+
+            election.evaluated = true;
```

Comment:
Maybe this could be put after the call of `_evaluateBallotRange`, in the chance that that method is overridden with some custom logic that checks for `election.evaluated` and it wouldn't be true because the last ballot batch is still not processed.

Reply:
Yeah, I was trying to save gas, but it makes too much noise, so I'll change it.

However, I'm going to try putting this in ElectionModule.sol, because it should be independent from the counting method, in the `evaluate()` function. Can you take a look?

Reply:
Oh yeah, makes more sense now :)